File: 02_Object_Oriented_Programming/03_Constructor_and_It's_Types/03_Copy_Constructor/javascript/movie.js
class Movie {
    // Parameterized constructor
    constructor(titleOrMovie, duration) {
        if (titleOrMovie instanceof Movie) {
            // Copy constructor
            this.title = titleOrMovie.title;
            this.duration = titleOrMovie.duration;
        } else {
            this.title = titleOrMovie;
            this.duration = duration;
        }
    }

    displayDetails() {
        console.log(`Title: ${this.title}, Duration: ${this.duration} mins`);
    }
}

module.exports = Movie;

File: 02_Object_Oriented_Programming/03_Constructor_and_It's_Types/03_Copy_Constructor/python/movie.py
import copy


class Movie:
    # Parameterized constructor
    def __init__(self, title, duration):
        self.title = title
        self.duration = duration

    # Copy constructor equivalent using class method
    @classmethod
    def from_movie(cls, other):
        return cls(other.title, other.duration)

    # Python also supports copy via __copy__
    def __copy__(self):
        return Movie(self.title, self.duration)

    def display_details(self):
        print(f"Title: {self.title}, Duration: {self.duration} mins")

